src/main/java/com/google/api/generator/spring/composer/SpringAutoConfigClassComposer.java

```diff
@@ -350,17 +350,75 @@ private static MethodDefinition createTransportChannelProviderBeanMethod(
             .addDescription(nameStringAssignmentExpr)
             .build();
 
-    return MethodDefinition.builder()
-        .setHeaderCommentStatements(
-            SpringAutoconfigCommentComposer.createTransportChannelProviderComment())
-        .setName(methodName)
-        .setScope(ScopeNode.PUBLIC)
-        .setReturnType(STATIC_TYPES.get("TransportChannelProvider"))
-        .setAnnotations(
-            Arrays.asList(
-                AnnotationNode.withType(STATIC_TYPES.get("Bean")), conditionalOnMissingBean))
-        .setReturnExpr(returnExpr)
-        .build();
+    MethodDefinition.Builder beanMethodBuilder =
+        MethodDefinition.builder()
+            .setHeaderCommentStatements(
+                SpringAutoconfigCommentComposer.createTransportChannelProviderComment())
+            .setName(methodName)
+            .setScope(ScopeNode.PUBLIC)
+            .setReturnType(STATIC_TYPES.get("TransportChannelProvider"))
+            .setAnnotations(
+                Arrays.asList(
+                    AnnotationNode.withType(STATIC_TYPES.get("Bean")), conditionalOnMissingBean));
+
+    // LanguageServiceSettings.defaultTransportChannelProvider()
+    MethodInvocationExpr defaultTransportChannelProviderExpr =
+        MethodInvocationExpr.builder()
+            .setMethodName("defaultTransportChannelProvider")
+            .setStaticReferenceType(types.get("ServiceSettings"))
+            .setReturnType(STATIC_TYPES.get("TransportChannelProvider"))
+            .build();
+
+    // LanguageServiceSettings.defaultHttpJsonTransportProviderBuilder().build()
+    Expr defaultTransportProviderExprChain =
+        MethodInvocationExpr.builder()
+            .setStaticReferenceType(types.get("ServiceSettings"))
+            .setMethodName("defaultHttpJsonTransportProviderBuilder")
+            .build();
+    MethodInvocationExpr defaultHttpJsonTransportProviderExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(defaultTransportProviderExprChain)
+            .setMethodName("build")
+            .setReturnType(STATIC_TYPES.get("InstantiatingHttpJsonChannelProvider"))
+            .build();
```

Comment:
Since this section is only used inside the if block, it would be better to move it into the if block, you could create a separate private method to improve the readability if you think it makes the if block too long.

Reply:
Updated, thanks for the suggestion!